aiida/manage/configuration/profile.py

```diff
@@ -134,14 +134,27 @@ def get_repository_container(self) -> 'Container':
         """
         from disk_objectstore import Container
 
-        filepath = os.path.join(self.repository_path, 'container')
+        filepath = self._container_path
         container = Container(filepath)
 
-        if not container.is_initialised:
+        if not self.container_is_initialised:
             container.init_container(clear=True, **self.defaults['repository'])  # pylint: disable=unsubscriptable-object
 
         return container
 
+    @property
+    def container_is_initialised(self):
```

Comment:
I am wondering if we should remove `container` from these methods. I know that I called the method `get_repository_container` but the `container` part really comes from this particular implementation with the `disk-objectstore`. If we were to ever change it (don't think we will but in principle) the container concept may disappear. Maybe it would be better to have the following method and property names:
```
def get_repository() -> `Repository`
def is_repository_initialised
def initialise_repository
```
Looking at this more, currently `get_repository_container` returns the `Container` instance, but this is also a leaky abstraction. Some of the calls to this method currently just use it to get the UUID of the repo. Since then I added `Profile.get_repository_uuid` which is better and those places should use that instead.

Ok, this was a bit a train of thought and I am now thinking that we should just merge this and then I will do another PR with some proposed interface and name changes. What do you think?

Reply:
Sounds good to me. Then we can move on and you can disregard my comments since it should be updated anyway in the upcoming PR and they are not functionally critical.

Reply:
OK, then I will merge this now and include your comments in the next PR. Then we can start testing the migration again. Thanks guys